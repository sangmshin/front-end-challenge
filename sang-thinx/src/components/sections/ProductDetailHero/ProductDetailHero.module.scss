@import 'variables';

.root {
  background-color: $primary-lightgrey;
}