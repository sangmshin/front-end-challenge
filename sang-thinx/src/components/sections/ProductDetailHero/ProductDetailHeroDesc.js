import React from 'react';

const ProductDetailHeroDesc = () => (
  <div>
    <p>
      description goes here
    </p>
  </div>
);

export default ProductDetailHeroDesc;