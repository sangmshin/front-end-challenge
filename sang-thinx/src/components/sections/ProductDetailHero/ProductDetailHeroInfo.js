import React from 'react';
import { array, shape, string } from 'prop-types';
import styles from './ProductDetailHeroInfo.module.scss';
import {
  ColorSelector,
  QuantitySelector,
  Link,
  Dropdown,
  // DropdownButton
} from 'components/elements';
import { SIZE_CHART_TEXT } from 'core/constants';

const ProductDetailHeroInfo = ({
  product_title: title,
  product_droplets: droplets,
  description,
  price,
  available_colors: availableColors,
  available_sizes: availableSizes,
  size_chart: sizeChart, 
  savings,
}) => {
  return (
    <div className={styles.root}>
      <div className={styles['info-container']}>
        <h1 className={styles.title}>{title}</h1>
        <p className={styles.droplets}>{droplets}</p>
        <p className={styles.description}>{description}</p>
      </div>
      <div className={styles['options-container']}>
        <h2 className={styles.price}>${price}</h2>
        <ColorSelector colors={availableColors} />
        <QuantitySelector />
        <Dropdown
          className={styles['size-dropdown']}
          items={availableSizes}
          type='size'
        />
        {sizeChart &&
          <Link className={styles['size-chart']}>{SIZE_CHART_TEXT}</Link>
        }
      </div>
    </div>
  )
};

ProductDetailHeroInfo.propTypes = {
  product_title: string.isRequired,
  product_droplets: string.isRequired,
  description: string.isRequired,
  price: string.isRequired,
  available_colors: array.isRequired,
  available_sizes: array.isRequired,
  savings: shape({
    label: string,
    url: string
  })
}

ProductDetailHeroInfo.defaultProps = {
  savings: null
}

export default ProductDetailHeroInfo;