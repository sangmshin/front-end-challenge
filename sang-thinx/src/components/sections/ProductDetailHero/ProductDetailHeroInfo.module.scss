@import 'variables';

.root {
  padding: 90px 0 40px;

  .info-container > * {
    &:not(:last-child) {
      margin-bottom: 20px;
    }
  }

  .title {
    font-size: 0.8125rem;
    text-transform: uppercase;
  }

  .subtitle, .description {
    font-size: 0.75rem;
    line-height: 1.375rem;
  }
}

@media (min-width: $screen-md) {
  .root {
    padding: 30px 0;

    .info-container > * {
      &:not(:last-child) {
        margin-bottom: 30px;
      }
    }

    .title {
      font-size: 0.875rem;
    }

    .subtitle, .description {
      font-size: 1.125rem;
      line-height: 2rem;
    }
  }
}

@media (min-width: $screen-lg) {
  .root {
    display: flex;
    flex-direction: row-reverse;
    align-items: center;
    position: sticky;
    position: -webkit-sticky; /* Safari */
    top: 0px;
    width: 25%;
    height: 100vh;
    padding: 0;

    .info-container {
      max-width: 340px;
    }
  }
}
