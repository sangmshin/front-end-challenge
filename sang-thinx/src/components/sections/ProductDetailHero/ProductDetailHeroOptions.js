import React from 'react';
import { array, bool, shape, string } from 'prop-types';
import styles from './ProductDetailHeroOptions.module.scss';
import {
  Button,
  ColorSelector,
  Dropdown,
  Link,
  QuantitySelector,
} from 'components/elements';
import { ADD_TO_CART_TEXT, SIZE_CHART_TEXT } from 'core/constants';

const ProductDetailHeroOptions = ({
  price,
  available_colors: availableColors,
  available_sizes: availableSizes,
  size_chart: sizeChart, 
  savings,
}) => (
  <div className={styles.root}>
    <div>
      <h2 className={styles.price}>${price}</h2>
      <ColorSelector colors={availableColors} />
      <QuantitySelector max={9} />
      <Dropdown
        className={styles['size-dropdown']}
        items={availableSizes}
        type='size'
      />
      {sizeChart &&
        <Link className={styles['size-chart']}>
          {SIZE_CHART_TEXT}
        </Link>
      }
      <Button
        accessibleText={ADD_TO_CART_TEXT}
        ariaLabel={ADD_TO_CART_TEXT}
        className={styles['add-to-cart']}
        type='submit'
      >
        {ADD_TO_CART_TEXT}
      </Button>
      {savings &&
        <Link className={styles.savings}>
          {savings.label}
        </Link>
      }
    </div>
  </div>
);

ProductDetailHeroOptions.propTypes = {
  price: string.isRequired,
  available_colors: array.isRequired,
  available_sizes: array.isRequired,
  size_chart: bool.isRequired,
  savings: shape({
    label: string,
    url: string
  })
}

ProductDetailHeroOptions.defaultProps = {
  savings: null
}

export default ProductDetailHeroOptions;