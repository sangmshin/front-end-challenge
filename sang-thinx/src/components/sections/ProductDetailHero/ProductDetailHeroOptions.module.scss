@import 'variables';

.root {
  padding: 30px 0;

  .options-container {
    text-transform: lowercase;
    
    .price {
      font-size: 1rem;
    }

    .size-dropdown {
      margin-top: 20px;
    }

    .size-chart {
      margin-top: 5px;
      font-size: 0.6875rem;
      text-decoration: underline;
    }

    .add-to-cart {
      display: block;
      width: 100%;
      padding: 7px;
      margin-top: 20px;
      background-color: $primary-black;
      color: $type-white;
      font-size: 0.8125rem;
      font-weight: bold;
    }

    .savings {
      font-size: 0.75rem;
      text-decoration: underline;
      text-align: center;
      width: 100%;
      margin-top: 25px;
    }
  }
}

@media (min-width: $screen-md) {
  .root {
    padding: 30px 0;
    width: 100%;
  }
}

@media (min-width: $screen-lg) {
  .root {
    display: flex;
    align-items: center;
    position: sticky;
    position: -webkit-sticky; /* Safari */
    top: 0px;
    width: 25%;
    height: 100vh;
    padding: 0;

    .options-container {
      width: 100%;
    }
  }
}

@media (min-width: $screen-xlg) {
  .root {
    flex-direction: row;
    
    .options-container {
      width: 100%;
      max-width: 310px;
    }
  }
}