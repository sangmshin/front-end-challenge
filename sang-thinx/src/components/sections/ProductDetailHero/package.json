{
  "name": "ProductDetailHero",
  "main": "./ProductDetailHero.js"
}
