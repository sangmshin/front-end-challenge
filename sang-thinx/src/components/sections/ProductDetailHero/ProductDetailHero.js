import React from 'react';
import Slider from 'react-slick';
import ProductDetailHeroDesc from './ProductDetailHeroDesc';
import styles from './ProductDetailHero.module.scss';

const ProductDetailHero = ({
  images,
  ...otherProps
}) => {
  const renderSlides = () => (
    images.map(({ src, alt }, index) => (
      <img src={src} alt={alt} key={index} />
    ))
  );
  const config = {
    dots: true,
    infinite: true,
    speed: 500,
    slidesToShow: 1,
    slidesToScroll: 1
  };

  return (
    <section className={styles.root}>
      {images &&
        <Slider {...config}>
          {renderSlides()}
        </Slider>
      }
      <ProductDetailHeroDesc {...otherProps} />
    </section>
  )
}

export default ProductDetailHero;