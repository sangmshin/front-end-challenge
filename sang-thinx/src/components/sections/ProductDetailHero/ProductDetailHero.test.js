import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ProductDetailHero } from './ProductDetailHero';
import { hero } from 'data/pageData.json';

const wrapper = shallow(
  <ProductDetailHero {...hero} />
);

describe('Component :: ProductDetailHero', () => {
  describe('#render', () => {
    it('should match snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
