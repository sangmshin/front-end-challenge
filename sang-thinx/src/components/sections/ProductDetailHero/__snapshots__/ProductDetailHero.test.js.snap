// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component :: ProductDetailHero #render should match snapshot 1`] = `
<section
  className="root page-gutter"
>
  <div
    className="product-detail-wrapper page-max-width"
  >
    <ProductDetailHeroInfo
      available_colors={
        Array [
          "black",
          "beige",
        ]
      }
      available_sizes={
        Array [
          "xsmall",
          "small",
          "medium",
          "large",
          "xlarge",
          "xxlarge",
        ]
      }
      description="The phrase \\"treat yo'self\\" reaches new belly-button brushing heights with the Hi-Waist, and it's here just in time for cuffing season."
      price="35.00"
      product_subtitle="Heavy days. Holds up to 2 tampon's worth"
      product_title="Hi - Waist"
      savings={
        Object {
          "label": "build a cycle set and save up to 20%",
          "url": "",
        }
      }
      size_chart={true}
    />
    <div
      className="product-images-container"
    >
      <div
        className="image-wrapper"
        key="0"
      >
        <img
          alt="Hi-Waist front 1"
          onClick={[Function]}
          role="button"
          src="https://thinx-assets.s3.amazonaws.com/highwaist_black_front_01_1024x1024.jpg"
        />
      </div>
      <div
        className="image-wrapper"
        key="1"
      >
        <img
          alt="Hi-Waist front 2"
          onClick={[Function]}
          role="button"
          src="https://thinx-assets.s3.amazonaws.com/highwaist_black_front_02_1024x1024.jpg"
        />
      </div>
      <div
        className="image-wrapper"
        key="2"
      >
        <img
          alt="Hi-Waist front 3"
          onClick={[Function]}
          role="button"
          src="https://thinx-assets.s3.amazonaws.com/highwaist_black_front_03_1024x1024.jpg"
        />
      </div>
      <div
        className="image-wrapper"
        key="3"
      >
        <img
          alt="Hi-Waist front 4"
          onClick={[Function]}
          role="button"
          src="https://thinx-assets.s3.amazonaws.com/highwaist_black_front_04_1024x1024.jpg"
        />
      </div>
      <div
        className="image-wrapper"
        key="4"
      >
        <img
          alt="Hi-Waist side"
          onClick={[Function]}
          role="button"
          src="https://thinx-assets.s3.amazonaws.com/highwaist_black_side_1024x1024.jpg"
        />
      </div>
      <div
        className="image-wrapper"
        key="5"
      >
        <img
          alt="Hi-Waist back"
          onClick={[Function]}
          role="button"
          src="https://thinx-assets.s3.amazonaws.com/highwaist_black_back_1024x1024.jpg"
        />
      </div>
    </div>
    <ProductDetailHeroOptions
      available_colors={
        Array [
          "black",
          "beige",
        ]
      }
      available_sizes={
        Array [
          "xsmall",
          "small",
          "medium",
          "large",
          "xlarge",
          "xxlarge",
        ]
      }
      description="The phrase \\"treat yo'self\\" reaches new belly-button brushing heights with the Hi-Waist, and it's here just in time for cuffing season."
      price="35.00"
      product_subtitle="Heavy days. Holds up to 2 tampon's worth"
      product_title="Hi - Waist"
      savings={
        Object {
          "label": "build a cycle set and save up to 20%",
          "url": "",
        }
      }
      size_chart={true}
    />
  </div>
</section>
`;
